### ISSUE TITLE
rke 0.1.17 Can't initiate NewClient: protocol not available

### ISSUE THREAD

**Issue by gregkeys** [rancher/rke#1189]
I updated from rke 0.1.15 and tried to update our rancher installation to kubernetes 1.11.8 and received the following error.  Downgrading from 0.1.17 back to 0.1.15 allows rke to connect again.

**RKE version:**
rke 0.1.17 on windows 10 professional

**Steps to Reproduce:**
```
$ rke up --config ~/rancher.yml
```

**Results:**
```
time="2019-03-08T20:31:04-08:00" level=info msg="Building Kubernetes cluster"
time="2019-03-08T20:31:04-08:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-03-08T20:31:04-08:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-03-08T20:31:04-08:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-03-08T20:31:04-08:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-03-08T20:31:04-08:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-03-08T20:31:04-08:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-03-08T20:31:04-08:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-03-08T20:31:04-08:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-03-08T20:31:04-08:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-03-08T20:31:04-08:00" level=fatal msg="Cluster must have at least one etcd plane host: failed to connect to the following etcd host(s) [172.x.x.x]"
```

**Comment by w-riley** [comment]
I'm seeing the same issue trying to deploy a new cluster. I'm also using Windows 10 Pro.

Last working version for me is v0.1.17-rc1

**Comment by Dan-Coleman** [comment]
Hi, can confirm I am also experiencing this issue on Windows 10 Enterprise.

Rolling back to 0.1.15 allows the connection to proceed.

Thanks

**Comment by fellipecm** [comment]
Same issue on Windows 10 Home and RKE 0.2.0 (I had the same issue with 0.1.7 before so I am assuming all versions greater than 0.1.15 are affected). Thanks!

**Comment by gregkeys** [comment]
Still gettting the same issue on 0.2.1

```
$ rke up --config ~/Configs/rancher/rancher-rke.yml
time="2019-04-01T19:59:53-07:00" level=info msg="Initiating Kubernetes cluster"
time="2019-04-01T19:59:53-07:00" level=info msg="[state] Possible legacy cluster detected, trying to upgrade"
time="2019-04-01T19:59:53-07:00" level=info msg="[reconcile] Rebuilding and updating local kube config"
time="2019-04-01T19:59:53-07:00" level=info msg="Successfully Deployed local admin kubeconfig at [C:\\Users\\nunya\\Configs\\rancher/kube_config_rancher-rke.yml]"
time="2019-04-01T19:59:54-07:00" level=info msg="[reconcile] host [172.x.x.x] is active master on the cluster"
time="2019-04-01T19:59:54-07:00" level=info msg="[state] Fetching cluster state from Kubernetes"
time="2019-04-01T19:59:54-07:00" level=info msg="[state] Successfully Fetched cluster state to Kubernetes ConfigMap: cluster-state"
time="2019-04-01T19:59:54-07:00" level=info msg="[certificates] Getting Cluster certificates from Kubernetes"
time="2019-04-01T19:59:57-07:00" level=info msg="[certificates] Creating service account token key"
time="2019-04-01T19:59:57-07:00" level=info msg="[certificates] Successfully fetched Cluster certificates from Kubernetes"
time="2019-04-01T19:59:57-07:00" level=info msg="Successfully Deployed state file at [C:\\Users\\nunya\\Configs\\rancher/rancher-rke.rkestate]"
time="2019-04-01T19:59:57-07:00" level=info msg="[certificates] Generating admin certificates and kubeconfig"
time="2019-04-01T19:59:57-07:00" level=info msg="Successfully Deployed state file at [C:\\Users\\nunya\\Configs\\rancher/rancher-rke.rkestate]"
time="2019-04-01T19:59:57-07:00" level=info msg="Building Kubernetes cluster"
time="2019-04-01T19:59:57-07:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-04-01T19:59:57-07:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-04-01T19:59:57-07:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-04-01T19:59:57-07:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-04-01T19:59:57-07:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-04-01T19:59:57-07:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-04-01T19:59:57-07:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-04-01T19:59:57-07:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-04-01T19:59:57-07:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-04-01T19:59:57-07:00" level=fatal msg="Cluster must have at least one etcd plane host: failed to connect to the following etcd host(s) [172.x.x.x]"



$ rke up --config ~/Configs/rancher/rancher-rke.yml
time="2019-04-01T20:00:44-07:00" level=info msg="Initiating Kubernetes cluster"
time="2019-04-01T20:00:44-07:00" level=info msg="[certificates] Generating admin certificates and kubeconfig"
time="2019-04-01T20:00:44-07:00" level=info msg="Successfully Deployed state file at [C:\\Users\\nunya\\Configs\\rancher/rancher-rke.rkestate]"
time="2019-04-01T20:00:44-07:00" level=info msg="Building Kubernetes cluster"
time="2019-04-01T20:00:44-07:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-04-01T20:00:44-07:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-04-01T20:00:44-07:00" level=info msg="[dialer] Setup tunnel for host [172.x.x.x]"
time="2019-04-01T20:00:44-07:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-04-01T20:00:44-07:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-04-01T20:00:44-07:00" level=warning msg="Failed to set up SSH tunneling for host [172.x.x.x]: Can't initiate NewClient: protocol not available"
time="2019-04-01T20:00:44-07:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-04-01T20:00:44-07:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-04-01T20:00:44-07:00" level=warning msg="Removing host [172.x.x.x] from node lists"
time="2019-04-01T20:00:44-07:00" level=fatal msg="Cluster must have at least one etcd plane host: failed to connect to the following etcd host(s) [172.x.x.x]"
```

**Comment by alena1108** [comment]
@galal-hussein most likely related to docker version update. Based on users' comments, the issue exits in v0.2 and later versions of v0.1 - where we bumped up norman version along with the docker version.

**Comment by Negashev** [comment]
repeat v0.2.2

**Comment by Negashev** [comment]
Work with ubuntu on windows 10
https://docs.microsoft.com/en-us/windows/wsl/install-win10

**Comment by GarbageYard** [comment]
I am using Windows 10 Enterprise, Version: 1803, OS build: 17134.648. RKE version: v0.1.18. I am using Ubuntu installed using Windows Subsystem for Linux.

Content of ```rancher-cluster.yml```

```
nodes:
  - address: 10.150.16.54
    internal_address: 10.150.16.54
    user: rke
    role: [controlplane,worker,etcd]
    ssh_key_path: ~/.ssh/id_rsa
  - address: 10.150.16.55
    internal_address: 10.150.16.55
    user: rke
    role: [controlplane,worker,etcd]
    ssh_key_path: ~/.ssh/id_rsa
  - address: 10.150.16.56
    internal_address: 10.150.16.56
    user: rke
    role: [controlplane,worker,etcd]
    ssh_key_path: ~/.ssh/id_rsa
    
services:
  etcd:
    snapshot: true
    creation: 6h
    retention: 24h
```

**Error:**
```
./rke.exe up --config ./rancher-cluster.yml
time="2019-05-03T11:43:53+05:30" level=info msg="Building Kubernetes cluster"
time="2019-05-03T11:43:53+05:30" level=info msg="[dialer] Setup tunnel for host [10.150.16.54]"
time="2019-05-03T11:43:54+05:30" level=warning msg="Failed to set up SSH tunneling for host [10.150.16.54]: Can't initiate NewClient: protocol not available"
time="2019-05-03T11:43:54+05:30" level=info msg="[dialer] Setup tunnel for host [10.150.16.55]"
time="2019-05-03T11:43:54+05:30" level=warning msg="Failed to set up SSH tunneling for host [10.150.16.55]: Can't initiate NewClient: protocol not available"
time="2019-05-03T11:43:54+05:30" level=info msg="[dialer] Setup tunnel for host [10.150.16.56]"
time="2019-05-03T11:43:54+05:30" level=warning msg="Failed to set up SSH tunneling for host [10.150.16.56]: Can't initiate NewClient: protocol not available"
time="2019-05-03T11:43:54+05:30" level=warning msg="Removing host [10.150.16.54] from node lists"
time="2019-05-03T11:43:54+05:30" level=warning msg="Removing host [10.150.16.55] from node lists"
time="2019-05-03T11:43:54+05:30" level=warning msg="Removing host [10.150.16.56] from node lists"
time="2019-05-03T11:43:54+05:30" level=fatal msg="Cluster must have at least one etcd plane host: failed to connect to the following etcd host(s) [10.150.16.54]"
```

**Comment by robkaandorp** [comment]
> Work with ubuntu on windows 10
> https://docs.microsoft.com/en-us/windows/wsl/install-win10

I can verify, this indeed works with version 0.1.18, but you will have to use the linux version of rke.

Start an Ubuntu shell in windows, go to the correct drive and path (you will find your C: drive under /mnt/c) and run `./rke_linux_amd64 up`.

**Comment by hbudde** [comment]
Any progress on this topic? I am running rke v0.2.8 on Windows 10 Pro Version 1903 to provisioning a cluster with RancherOS installed and having exactly the same problem (Can't initiate NewClient: protocol not available). What exactly does this mean? I do not want to install a Linux subsystem on my local maschine. Is there any workaround? I already had a look at https://raw.githubusercontent.com/rancher/rke/26e2979d20c0bc04b23bdad1eba70ce451bb0fff/hosts/tunnel.go. What I understand so far is that the error happens when rke tries to connect to Docker on the host.

**Comment by superseb** [comment]
Seems we should be able to remove the config that is causing this issue, I created a PR for review and testing

**Comment by bmdepesa** [comment]
rke v0.3.0-rc9
Windows 10 Professional 1803 17134.885

* Reproduced with RKE v0.1.17
```
c:\Dev\tools>rke_windows-amd64-0117.exe up --config c:\dev\rke-yml\ec2-simple.yml
[INFO] Building Kubernetes cluster
[INFO] [dialer] Setup tunnel for host [x.x.x.x]
[WARN] Failed to set up SSH tunneling for host [x.x.x.x]: Can't initiate NewClient: protocol not available
[INFO] [dialer] Setup tunnel for host [x.x.x.x]
[WARN] Failed to set up SSH tunneling for host [x.x.x.x]: Can't initiate NewClient: protocol not available
[INFO] [dialer] Setup tunnel for host [x.x.x.x]
[WARN] Failed to set up SSH tunneling for host [x.x.x.x]: Can't initiate NewClient: protocol not available
```
* Resolved in v0.3.0-rc9
```
c:\Dev\tools>rke_windows-amd64-030rc9.exe up --config c:\Dev\rke-yml\ec2-simple.yml
time="2019-09-06T10:45:20-07:00" level=warning msg="This is not an officially supported version (v0.3.0-rc9) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest"
time="2019-09-06T10:45:20-07:00" level=info msg="Running RKE version: v0.3.0-rc9"
time="2019-09-06T10:45:20-07:00" level=info msg="Initiating Kubernetes cluster"
time="2019-09-06T10:45:20-07:00" level=info msg="[dialer] Setup tunnel for host [x.x.x.x]"
time="2019-09-06T10:45:20-07:00" level=info msg="[dialer] Setup tunnel for host [x.x.x.x]"
time="2019-09-06T10:45:20-07:00" level=info msg="[dialer] Setup tunnel for host [x.x.x.x]"
time="2019-09-06T10:45:21-07:00" level=info msg="Checking if container [cluster-state-deployer] is running on host [x.x.x.x], try #1"
...
```
* Cluster provisions successfully
